### DISCUSSION TITLE
Shortcut for calling a function to toggle padding

### DISCUSSION THREAD

**Discussion by frankjonen** [wezterm/wezterm#5854] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Mutter

### WezTerm version

20240722_181403_745518dc

### Ask your question!

I’m looking for some pointers on how I’d go about creating a shortcut binding for a padding toggle.

I’ve tried to wrap the `config.window_padding` into a `local function` but triggering with `action = run function_name()` did nothing. 

Any ideas?

**Comment by nekowinston** [comment]
Look into [`set_config_overrides`](https://wezfurlong.org/wezterm/config/lua/window/set_config_overrides.html) and [`action_callback`](https://wezfurlong.org/wezterm/config/lua/wezterm/action_callback.html).
I've also posted [an example](https://github.com/wez/wezterm/discussions/3839#discussioncomment-6123338) on how to automatically toggle the padding depending on which foreground process is active.

**Comment by frankjonen** [comment]
Thanks for the example, I’ll definitely try that out for making wordgrinder easier on the eyes. 

For my toggle issue, I took some of yours and tried to get it done that way but now I’m getting a window error when saving the config. Which tells me I’ve done this wrong. 

```lua

config.keys = {
  {
    key = 'w',
    mods = 'SHIFT|CTRL',
    action = window:set_config_overrides({
      window_padding = { left = 250, right = 250, top = 0, bottom = 0 },
    }),
  }
}
```

**Reply by nekowinston** [reply, marked as answer]
You need to use `action_callback`, which provides the `window` object for set_config_overrides; e.g. like this:
```lua
config.keys = {
	{
		key = "w",
		mods = "SHIFT|CTRL",
		action = wezterm.action_callback(function(window, pane)
			wezterm.GLOBAL.zen = not wezterm.GLOBAL.zen

			if wezterm.GLOBAL.zen then
				window:set_config_overrides({
					window_padding = { left = 250, right = 250, top = 0, bottom = 0 },
				})
			else
				window:set_config_overrides({
					window_padding = { left = 0, right = 0, top = 0, bottom = 0 },
				})
			end
		end),
	},
}
```

**Reply by frankjonen** [reply]
Thank you so much for writing out the whole thing. I would have never guessed all that zen stuff.

**Reply by nekowinston** [reply]
`zen` is just a boolean value stored in the WezTerm process (https://wezfurlong.org/wezterm/config/lua/wezterm/GLOBAL.html), the name could be anything you want. I went with `zen` since zen modes usually have padding like this.
